std/http/server.ts

```diff
@@ -483,7 +478,7 @@ interface ServerConfig {
  *       req.respond({ body });
  *     }
  */
-export function serve(addr: string | ServerConfig): Server {
+export function serve(addr: string | Deno.ListenOptions): Server {
```

Comment:
:+1: can you apply the same rule to `serveTLS` and  `listenAndServeTLS`?

Reply:
Are you sure?  From the code:

```ts
export type HTTPSOptions = Omit<Deno.ListenTLSOptions, "transport">;
```

It is exported, and it is different that `Deno.ListenTLSOptions` intentionally.

Reply:
You're right, I see it skips "protocol" field - then I guess it makes sense to apply the same for `serve`.

Reply:
yeah, let me align them...